build-support/common.sh

```diff
@@ -67,3 +67,13 @@ function git_merge_base() {
   git rev-parse --symbolic-full-name --abbrev-ref HEAD@{upstream} 2>/dev/null || echo 'master'
 
 }
+
+function validate_executable() {
+  local -r exe_name="$1"
+  if ! hash "$exe_name"; then
```

Comment:
I think the most portable technique is `command -v`, but that was a few weeks ago I checked so this could work too.

Reply:
Do you have a source for that? I would love to use it if so (really!), but I have literally never seen that before while I have seen `hash` (and `hash` is POSIX iirc). Are we using `command -v` elsewhere in this repo?

Reply:
https://stackoverflow.com/questions/592620/how-to-check-if-a-program-exists-from-a-bash-script?answertab=votes#tab-top

We will be replacing all uses of `which` with `command -v` once I finish the shellcheck PR (which depends on https://github.com/pantsbuild/pants/pull/7633 + me finishing this essay first).

Reply:
Oh wow, I was completely backwards! Thank you!!